src/transforms/groupby.js

```diff
@@ -126,31 +128,28 @@ function transformOne(trace, state, attributeSet) {
     var newData = new Array(groupNames.length);
     var len = groups.length;
 
-    var style = opts.style || {};
+    var arrayAttrs = Lib.findArrayAttributes(trace);
 
-    var arrayAttributes = attributeSet
-        .filter(function(array) {return Array.isArray(Lib.nestedProperty(trace, array).get());});
+    var style = opts.style || {};
 
     for(var i = 0; i < groupNames.length; i++) {
         var groupName = groupNames[i];
 
-        // TODO is this the best pattern ???
-        // maybe we could abstract this out
-        var newTrace = newData[i] = Lib.extendDeep({}, trace);
+        var newTrace = newData[i] = Lib.extendDeepNoArrays({}, trace);
 
-        arrayAttributes.forEach(initializeArray.bind(null, newTrace));
+        arrayAttrs.forEach(initializeArray.bind(null, newTrace));
 
         for(var j = 0; j < len; j++) {
             if(groups[j] !== groupName) continue;
 
-            arrayAttributes.forEach(pasteArray.bind(0, newTrace, trace, j));
+            arrayAttrs.forEach(pasteArray.bind(0, newTrace, trace, j));
         }
 
         newTrace.name = groupName;
 
-        //  there's no need to coerce style[groupName] here
+        // there's no need to coerce style[groupName] here
         // as another round of supplyDefaults is done on the transformed traces
-        newTrace = Lib.extendDeep(newTrace, style[groupName] || {});
+        newTrace = Lib.extendDeepNoArrays(newTrace, style[groupName] || {});
```

Comment:
@monfera no one to extend arrays here, right?

This could be a solid performance boost.